test/e2e/framework/test_context.go

```diff
@@ -436,4 +437,46 @@ func AfterReadingAllFlags(t *TestContextType) {
 		}
 		os.Exit(1)
 	}
+
+	// Check for num-nodes if not set by looking for all nodes that are schedulable
+	if t.CloudConfig.NumNodes == -1 {
+		numNodes, err := countScheduleableNodes(t.KubeContext, t.KubeConfig, t.Host)
+		if err != nil {
+			klog.Errorf("Error getting num-nodes from API: %v", err)
+		} else {
+			fmt.Println("numNodes: ", numNodes)
+			t.CloudConfig.NumNodes = numNodes
+		}
+	}
+}
+
+// Gets a config used to create a client
+// TODO(stevesloka) We should simplify this across the framework and not duplicate
+func loadConfig(kubeContext, kubeconfig, host string) (*restclient.Config, error) {
+	c, err := RestclientConfig(kubeContext)
+	if err != nil {
+		if kubeconfig == "" {
+			return restclient.InClusterConfig()
+		}
+		return nil, err
+	}
+	return clientcmd.NewDefaultClientConfig(*c, &clientcmd.ConfigOverrides{ClusterInfo: clientcmdapi.Cluster{Server: host}}).ClientConfig()
+}
+
+// Counts the number of schedulable nodes without "NoSchedule" taints
+func countScheduleableNodes(kubeContext, kubeconfig, host string) (int, error) {
+	// Get Config
+	config, err := loadConfig(kubeContext, kubeconfig, host)
```

Comment:
we might want to use this function?
https://github.com/kubernetes/kubernetes/blob/b67126e350fdba6d89a0ee64310cac7db14e33d3/test/e2e/framework/util.go#L2120